lib/internal/modules/cjs/loader.js

```diff
@@ -567,7 +570,16 @@ function tryModuleLoad(module, filename) {
 }
 
 Module._resolveFilename = function(request, parent, isMain, options) {
-  if (NativeModule.nonInternalExists(request)) {
```

Comment:
Re-post as this seems to be buried from the timeline:

Maybe the correct place to insert this logic is `NativeModule.nonInternalExists` instead of `Module._resolveFilename`, also `NativeModule.nonInternalExists` is even less public anyways (you'd need `--expose-internals` to get access to `NativeModule`). You can change it to something like `NativeModule.getUnifiedId` which returns `undefined` if there is no such module with or without the prefix. That would also fix up `Module._resolveLookupPaths`.

It could also be faster if `NativeModule.getUnifiedId` simply queries a map of `key => { unified_id,   source }`with both `@nodejs/something` and `something` inserted as keys pointing to the same entry. Right now `NativeModule.nonInternalExists` does `NativeModule._source.hasOwnProperty(id)` in the end but `NativeModule._source` is entirely internal so we are free to change its structure. That'll avoid the extra overhead mentioned in the other review.

Reply:
Happy to update this, where would I find the source for NativeModule?